Gems/Atom/RHI/Code/Source/RHI/FrameGraphExecuter.cpp

```diff
@@ -97,12 +95,14 @@ namespace AZ
             AZ_Assert(group.IsComplete(), "Ending a context group before all child contexts have ended!");
             group.EndInternal();
             group.m_isSubmittable = true;
```

Comment:
This seems wrong but since I am not as familiar with Taskgraph so I may be wrong. Can you confirm that with your changes what will TaskGraph do? 

If lets say you have 3 groups that are executed in parallel where group 2 got added then 3 and then 1. Lets say group 3 finishes at time n, then group 2 finishes at time n+3 and then group 1 finishes at time n+5 and ExecuteGroupInternal takes 1 time for all the groups. Currently it will do the following

- Group 3  will finish at time n and go into a spin loop
- Group 2 will finish at time n+3, 
- Group 2 calls execute at time n+3
- Group 3 calls execute at time n+4
- Group 1 will finish at time n+ 5 
- Group 1 calls execute at time n+5. 

To me it seems like with TaskGraph it will do the following.

- Group 3 finishes at time n
- Group 2 fill finish at time n+3
- Group 1 will finish at time n+5
- Group 2 calls execute at time n+5
- Group 3 calls execute at time n+6
- Group 1 calls execute at time n+7
- 
  It seems to be that because you took out m_pendingGroups it is not in-efficient. Let me know if I am incorrect here
  
  
These are some very serious core changes that can have bugs which may not show up until later. Please add a cvar so that we can switch back and forth between the old way and new way to compare if new bugs related to perf/artifacts start showing up further down the road.

Reply:
So there are two problems here. 
- One is the problem that you cant execute in parallel which you can currently and this seems like a regression
- Second is order. However I do think that with task graph it will maintain order. So the execution order will be 2, 3, 1 which is correct as that is how they were added into m_groups

My recommendation would be to not convert this specific code to task graph until we figure out a way to execute in parallel. Maybe we can add a TODO for now.

Reply:
Which part do you think we aren't doing in parallel? 
The only bit that moved was the call to ExecuteGroupInternal, which was previously wrapped in a mutex so was serial only.

Reply:
So if you look at my example above. With the current way we finish all work by n+6 time whereas with the new approach we finish at n+8 time hence it is in-efficient. Basically Group 3 is able to call execute while group 1 is still encoding hence doing it in parallel with group 1.

Reply:
I reverted the PendingGroups change, FrameGraph::Execution has been moved back to Jobs.